types/plan/plan.go

```diff
@@ -177,15 +178,57 @@ func (output *KubernetesOutput) Merge(newoutput KubernetesOutput) {
 
 // Inputs defines the input section of plan
 type Inputs struct {
+	AbsRootDir          string                                   `yaml:"absRootDir"`
+	RelRootDir          string                                   `yaml:"relRootDir"`
 	RootDir             string                                   `yaml:"rootDir"`
 	K8sFiles            []string                                 `yaml:"kubernetesYamls,omitempty"`
 	QACaches            []string                                 `yaml:"qaCaches,omitempty"`
 	Services            map[string][]Service                     `yaml:"services"`                      // [serviceName][Services]
 	TargetInfoArtifacts map[TargetInfoArtifactTypeValue][]string `yaml:"targetInfoArtifacts,omitempty"` //[targetinfoartifacttype][List of artifacts]
 }
 
+// SetRootDir sets the path to the source directory in the plan.Spec.Inputs
+func (inputs *Inputs) SetRootDir(path string) error {
+	path = filepath.Clean(path)
+	inputs.RootDir = path
+
+	if filepath.IsAbs(path) {
+		inputs.AbsRootDir = path
+		currDir, err := os.Getwd()
+		if err != nil {
+			log.Errorf("Failed to get the current working directory. Error %q", err)
+			return err
+		}
+		relPath, err := filepath.Rel(currDir, path)
+		if err != nil {
+			log.Errorf("Failed to make the source directory %q relative to the current working directory %q Error %q", path, currDir, err)
+			return err
+		}
+		inputs.RelRootDir = relPath
+	} else {
+		inputs.RelRootDir = path
+		absPath, err := filepath.Abs(path)
+		if err != nil {
+			log.Errorf("Failed to get the absolute path for the source directory %q Error %q", path, err)
+			return err
+		}
+		inputs.AbsRootDir = absPath
+	}
+
+	return nil
+}
+
+// CICDSpec contains information specific to creating the CI/CD pipeline.
+type CICDSpec struct {
+	GitRepoDir    string
+	GitRepoURL    string
+	GitRepoBranch string
+	TargetPath    string
+}
```

Comment:
The type is meant to contain all extra information needed specifically for CICD.
Currently git repo details are the only extra info we need but that might change.

Reply:
If we add more fields we should name the fields based on what it stores. Essentially any data structure name should describe what data it contains and not what it will be used for.

Reply:
Done